src/rules/maxFileLineCountRule.ts

```diff
@@ -55,17 +54,13 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const ruleFailures: Lint.RuleFailure[] = [];
-        const ruleArguments = this.getOptions().ruleArguments;
-        const lineLimit: number = ruleArguments[0];
+        const lineLimit: number = this.ruleArguments[0];
         const lineCount: number = sourceFile.getLineStarts().length;
-        const disabledIntervals = this.getOptions().disabledIntervals;
-
-        if (lineCount > lineLimit && disabledIntervals.length === 0) {
-            const errorString = Rule.FAILURE_STRING_FACTORY(lineCount, lineLimit);
-            ruleFailures.push(new Lint.RuleFailure(sourceFile, 0, 1, errorString,
-              this.getOptions().ruleName));
+        if (lineCount <= lineLimit) {
+            return [];
         }
-        return ruleFailures;
+
+        return this.applyWithFunction(sourceFile, (ctx) =>
+            ctx.addFailure(0, 1, Rule.FAILURE_STRING(lineCount, lineLimit)));
```

Comment:
I like this better than constructing `new Lint.RuleFailure` directly because I don't have to pass in e.g. `ruleName` explicitly. Could have a `Rule#createFailure` method but I think that would only be used here.